avocado/core/varianter.py

```diff
@@ -447,132 +381,78 @@ def _skip_new_data_check(self, function, args):
             raise RuntimeError("Varianter already parsed, unable to execute "
                                "%s%s" % (function, args))
 
-    def data_inject(self, name, key, value, path=None):   # pylint: disable=E0202
+    def add_default_param(self, name, key, value, path=None):   # pylint: disable=E0202
         """
-        Inject entry to the mux tree (params database)
+        Stores the path/key/value into default params
 
+        :param name: Name of the component which injects this param
         :param key: Key to which we'd like to assign the value
         :param value: The key's value
         :param path: Optional path to the node to which we assign the value,
                      by default '/'.
         """
-        if self._skip_new_data_check("data_inject", (key, value, path)):
+        if self._skip_new_data_check("add_default_param", (key, value, path)):
             return
         if path is None:
             path = "/"
         if name not in self.default_params:
-            self.default_params[name] = tree.TreeNode()
+            self.default_params[name] = self.node_class()
         self.default_params[name].get_node(path, True).value[key] = value
 
-    def data_merge(self, tree):     # pylint: disable=E0202
+    def add_variants_plugin(self, plugin):     # pylint: disable=E0202
```

Comment:
Well this function is replaced with dispatcher's entry point. So at this point I think it is quite important... I mean yes, I can just forget this step existed and immediatelly start with dispatcher-based plugin initialization, but that would be even more complex patch. Also note that this patch existed months ago before I started working on making it properly supported as a plugin interface. That's the whole problem of this PR, I change things because they were not reviewed and merged until I developed another stage which means there is 3+ moths old code combined with new requirements. In a pure agile world, all of those changes would already been applied. In the old days this would only existed in my tree and I'd send a huge update which would remove the previous and add the new iterface (which would not be possible to review, that's why people started to use patch series and due to their length they started to use agile...)

Anyway with couple of hours of (in my eyes) wasted time I can rework this to only include the `dispatcher`-based version. But I'd really prefer using that time to finish the variant-id-based-names, internal filters and selftests instead.